### PR TITLE
Refactor

### PR DESCRIPTION
**Pull request by triopium**
Complete rewrite of openmedia-reduce.

### CODE REVIEW

**Comment by 4e1e0603** [comment]
- [xmlpipe](https://github.com/czech-radio/openmedia-reduce/blob/refactor/internal/xmlpipe.go)

  - <https://github.com/czech-radio/openmedia-reduce/blob/750306aae5794679549356fce229813014347586/internal/xmlpipe.go#L31C6-L31C29>
    change `PipeRundownheaderAmmend` to `PipeRundownHeaderAmmend`
    add missing documentation comment

  - <https://github.com/czech-radio/openmedia-reduce/blob/750306aae5794679549356fce229813014347586/internal/xmlpipe.go#L57C6-L57C23>
    `PipeRundownMinfiy`, misspelling
     add missing documentation comment

  - <https://github.com/czech-radio/openmedia-reduce/blob/750306aae5794679549356fce229813014347586/internal/xmlpipe.go#L57C6-L57C23>
    add missing documentation comment

- [xml](https://github.com/czech-radio/openmedia-reduce/blob/refactor/internal/xml.go)
    add missing documentation comment

- [shared.go](https://github.com/czech-radio/openmedia-reduce/blob/refactor/internal/shared.go#L55C6-L55C25)
   add missing documentation comment

- [shared.go](https://github.com/czech-radio/openmedia-reduce/blob/750306aae5794679549356fce229813014347586/internal/shared.go#L55C6-L55C25)
    (Maybe) be consistent and use abbreviation for RAM
    change  `DirectoryCreateInRam` to  `DirectoryCreateInRAM`

- [rundown_model](https://github.com/czech-radio/openmedia-reduce/blob/refactor/internal/rundown_model.go)
  add missing comments
  remove unused (commented) code

- [helpers.go](https://github.com/czech-radio/openmedia-reduce/blob/refactor/internal/helpers.go)
  add missing comments

Write a proper sentences: `// SetLogLevel: sets log level, default=0`


See <https://tip.golang.org/doc/comment>

File: internal/helpers_test.go [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+		}
+	}()
+	return com, wg
+}
+
+// TestMain setup, run tests, and teadrdown (cleanup after tests)
+func TestMain(m *testing.M) {
+	// TESTS SETUP
+
+	//// Setup logging
```
[the remark this answers is not in the document]
Comment (triopium) [reply internal/helpers_test.go:45]: To denote second level header.

File: internal/process.go [hunk 2]
```diff
@@ -0,0 +1,335 @@
...
+	WG      *sync.WaitGroup
+}
+
+type ProcessOptions struct {
+	SourceDirectory        string
+	DestinationDirectory   string
+	InputEncoding          string
+	OutputEncoding         string
+	ValidateWithDefaultXSD bool   // validate with bundled file
+	ValidateWithXSD        string // path to XSD file
```
[the remark this answers is not in the document]
Comment (triopium) [reply internal/process.go:37]: Should probably be ValidateWithXSDfile

File: internal/process.go [hunk 3]
```diff
@@ -0,0 +1,335 @@
...
+func (p *Process) DestroyWorkers() {
+	if p.Workers == nil {
+		return
+	}
+	for _, worker := range p.Workers {
+		worker.ArchiveWriter.Close()
+	}
+}
+
+func RundownFileNameNormalize(fileName string) {
```
[the remark this answers is not in the document]
Comment (triopium) [reply internal/process.go:103]: Just working on it yet.

File: internal/process_test.go [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+	"testing"
+)
+
+func Test_RundownNameParse(t *testing.T) {
+	type Case struct {
+		input  string
+		result string
+		error  error
+	}
+	// regexPattern := `(\p{L}+\s?\p{L}*\s?\p{L}*)`
```
[the remark this answers is not in the document]
Comment (triopium) [reply internal/process_test.go:16]: WIP

File: internal/process.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (4e1e0603) [reply internal/process.go:37]: I agree

File: internal/process_test.go (hunk 4 above)
[the remark this answers is not in the document]
Comment (4e1e0603) [reply internal/process_test.go:16]: OK, so add some comment e.g `TODO ...`

File: internal/process.go (hunk 3 above)
[the remark this answers is not in the document]
Comment (4e1e0603) [reply internal/process.go:103]: OK, so add some comment e.g TODO ...

**Pull request merged**